common/include/common/common_utils.hpp

```diff
@@ -442,6 +504,92 @@ static inline std::vector<reduction_param_t> get_reduction_params(Args& args) {
   }
 }
 
+/**
+ * @fn get_symm_params
+ * @brief Returns a vector containing the symm benchmark parameters, either
+ * read from a file according to the command-line args, or the default ones.
+ */
+template <typename scalar_t>
+static inline std::vector<symm_param_t<scalar_t>> get_symm_params(Args& args) {
+  if (args.csv_param.empty()) {
+    warning_no_csv();
+    std::vector<symm_param_t<scalar_t>> symm_default;
+    constexpr index_t dmin = 64, dmax = 1024;
+    constexpr scalar_t alpha{1};
+    constexpr scalar_t beta{1};
+    for (char side : {'l', 'r'}) {
+      for (char uplo : {'u', 'l'}) {
+        for (index_t m = dmin; m <= dmax; m *= 2) {
+          for (index_t n = dmin; n <= dmax; n *= 2) {
+              symm_default.push_back(
+                  std::make_tuple(side, uplo, m, n, alpha, beta));
+          }
+        }
+      }
+    }
+    return symm_default;
+  } else {
+    return parse_csv_file<symm_param_t<scalar_t>>(
+        args.csv_param, [&](std::vector<std::string>& v) {
+          if (v.size() != 6) {
+            throw std::runtime_error(
+                "invalid number of parameters (7 expected)");
```

Comment:
Thanks for your patch, @s-Nick!

```suggestion
                "invalid number of parameters (6 expected)");
```